hw2/04-api.js

```diff
@@ -1,7 +1,57 @@
-/** Exercise 04 - API **/
+/* eslint-disable no-console */
+/** Exercise 04 - API * */
 
 const url = 'https://restcountries.com/v3.1/all';
 
 // Add your code here
 
+// Function to help format large numbers
+const nf = Intl.NumberFormat();
+
+// Adds a single element to the dom
+const addToDom = (toPrint) => {
+  const results = document.getElementById('results');
+  const entry = document.createElement('li');
+  entry.appendChild(document.createTextNode(toPrint));
+  results.appendChild(entry);
+};
+
+// Function to add error message to screen
+const addErrorToDom = (toPrint) => {
+  const toGet = document.getElementById('results').parentElement;
+  console.log(toGet);
+  const results = document.createElement('div');
+  const entry = document.createElement('h1');
+  entry.style.color = 'red';
+  entry.textContent = toPrint;
+  results.appendChild(entry);
+  toGet.appendChild(results);
+};
+
+// Function that uses fetch to retrieve data and add it to the dom
+const getData = (uRL) => {
+  fetch(uRL)
+    .then((resp) => resp.json())
```

Comment:
I had no idea what "then()" does and had to look it up and it seems similar to async/await. I'm glad to see another way to handle things, thanks for the knowledge!